src/command/manager.js

```diff
@@ -16,7 +16,7 @@ class CommandManager {
     }
 
     execute(query) {
-        query = query.replace(":", "").trim().toLowerCase();
+        query = query.replace(COMMAND_PREFIX, "").trim().toLowerCase();
```

Comment:
I think a better solution is to add a prefix argument to the `CommandManager` constructor rather than using a global variable.

```js
 constructor(prefix, ...commands) {
     this.prefix = prefix;
 }
```

Reply:
I was thinking the same thing, but I'm not sure what a good way would be to inform the commands themselves about their prefix, especially without breaking any downstream extensions already using this library.

I had two potential solutions in mind (letting `CommandManager.constructor` inform the commands about the prefix, or don't burden commands with prefixes whatsoever and only handle that within `CommandManager`), but both would break compatibility AFAICT. What do you think?

Reply:
Hm, after thinking about it some more I feel like my concerns don't apply. A command currently does not really need to know the prefix, given that it is stripped out in command-manager beforehand anyway.

I'll update the PR to include your suggestions! 👍

Reply:
JavaScript unfortunately does not seem to support a combination of optional default parameters and rest parameters, i.e. something like `constructor(prefix = ":", ...commands)` is not backwards compatible right out of the box. I think one could implement a workaround by checking if `prefix` is just a string, and if it isn't, treat it like a command instead.

Would you rather have a break in backwards compatibility, or should I implement the mentioned workaround?

Reply:
There's no compatibility issue for my own search extensions, hahaha. However, I don't know how many third-party extension depends on this project. Maybe we need a semver to reduce such a breaking compatibility issue.

> letting CommandManager.constructor inform the commands about the prefix

> don't burden commands with prefixes whatsoever and only handle that within CommandManager

Sounds good. I hope we can achieve this without introducing extra complexity. 
IMO, adding a prefix argument should be the simplest solution if we forget the compatibility issue. 😆

Reply:
I have [implemented a backwards-compatible solution for the prefix](https://github.com/huhu/search-extension-core/pull/2/files#diff-ce8eda15db23293a57052703db5a7eb54ccf627f949e8e3d8d009e4ae8a800e1R2-R12), at least I think it should be compatible; a quick test seemed to work as intended if the first argument was a command: 

```javascript
constructor(prefixOrCommand, ...commands) {
    // The `prefixOrCommand` argument was introduced at a later point, which would break
    // backwards compatibility. To avoid this breakage the following code checks if the provided
    // `prefixOrCommand` is indeed a prefix, or if it is the first command and the default
    // prefix should be used instead.
    if (typeof prefixOrCommand === 'string') {
        this.prefix = prefixOrCommand;
    } else {
        this.prefix = ":";
        commands.unshift(prefixOrCommand);
    }
}
```

WDYT, is this fine?

Reply:
LGTM! 👍 Thanks a lot.